cli/operator/node.go

```diff
@@ -351,25 +364,31 @@ func setupGlobal(cmd *cobra.Command) (*zap.Logger, error) {
 	return zap.L(), nil
 }
 
-func setupDB(logger *zap.Logger, eth2Network beaconprotocol.Network) (*kv.BadgerDB, error) {
-	db, err := kv.New(logger, cfg.DBOptions)
+func setupDB(ctx context.Context, logger *zap.Logger, network beaconprotocol.BeaconNetwork, spDB *kv.BadgerDB) (*kv.BadgerDB, error) {
+	options := basedb.Options{
+		Path:       cfg.DBOptions.Path,
+		Reporting:  cfg.DBOptions.Reporting,
+		GCInterval: cfg.DBOptions.GCInterval,
+	}
+
+	db, err := kv.New(ctx, logger, options)
```

Comment:
As far as I know, passing context to a constructor is not a common pattern unless it's something specific. Instead, context is usually passed to methods that may be canceled. Could you please explain why we need to pass it here?